src/lib/fetch.ts

```diff
@@ -112,10 +113,25 @@ async function _handleRequest(
 ): Promise<any> {
   return new Promise((resolve, reject) => {
     fetcher(url, _getRequestParams(method, options, parameters, body))
-      .then((result) => {
+      .then(async (result) => {
         if (!result.ok) throw result
         if (options?.noResolveJson) return result
-        return result.json()
+        const json = await result.json()
+        let pagination: Pagination = {}
+        const links = result.headers.get('link')?.split(',')
+
+        if (links) {
+            links.forEach(link => {
+                const url = link.split(';')[0].replace(/[\<\>\s]/g, '')
+                const rel = JSON.parse(link.split(';')[1].replace(/[\<\>\s]/g, '').split('=')[1])
+                pagination[rel] = url
+            })
+            
+            if (!pagination.next) pagination.next = null
+            const total = result.headers.get('x-total-count') ?? '0'
+            pagination.total = parseInt(total)
+        }
+        return { ...json, ...pagination }
```

Comment:
This logic can be moved to the `listUsers` method instead. There is the option to return the `result` instead of `result.json()` by setting the `noResolveJson` option to true and then you should be able to handle destructuring the appropriate headers in there.

Reply:
Thanks for taking a look at this @kangmingtay!

I can't figure out how to grab the data by the time it gets back to `listUsers()`.

In `_handleRequest()` I can access the headers with `result.headers.get()`.
In `_request()` I have to use something like `data.headers.forEach()` because methods like `get` and `has` have been stripped.
Finally, by the time the data reaches `listUsers()`, `typeof` says `data` is an object, but I'm unable to access things the way I'm used to. Not to mention I'm not sure if I can even access the returned users at this point.

Maybe I've misunderstood your suggestion and gone down the wrong rabbit hole??

Reply:
In `_handleRequest()` it's simply returning the result
```js
if (options?.noResolveJson) return result
```
In `_request()` it's basically doing the same; spreading it out
```js
return options?.xform ? options?.xform(data) : { data: { ...data }, error: null }
```
And inside of `listUsers()`, my test of `console.log(data.json())` is throwing `data.json is not a function`.
```js
const { data, error } = await _request(this.fetch, 'GET', `${this.url}/admin/users`, {
  headers: this.headers,
  noResolveJson: true
})
if (error) throw error
console.log('list', data.json())
```

Reply:
@kangmingtay I was thinking about this at 4:45am, and realized what the issue is. We're not returning a response, but more like a copy of the response data.

The only way I can get this to work is to change what is returned in `_request()`. This way we're returning an actual response.

from this
```js
return options?.xform ? options?.xform(data) : { data: { ...data }, error: null }
```
to this
```js
return options?.xform ? options?.xform(data) : 
    options?.noResolveJson ? 
      new Response(data.body, { 
        headers: data.headers, 
        status: data.status, 
        statusText: data.statusText 
      }) 
      : { data: { ...data }, error: null }
```

Reply:
psh... also looks like I've managed to pull down upstream changes somehow. So there's a lot more diffs that what I've actually done for this.

Reply:
> psh... also looks like I've managed to pull down upstream changes somehow. So there's a lot more diffs that what I've actually done for this.

Got that rectified.